projects/cdk/directives/copy-processor/copy-processor.directive.ts

```diff
@@ -0,0 +1,25 @@
+import {Directive, HostListener, Inject, Input} from '@angular/core';
+import {WINDOW} from '@ng-web-apis/common';
+import {tuiDefaultProp} from '@taiga-ui/cdk/decorators';
+import {TuiStringHandler} from '@taiga-ui/cdk/types';
+import {identity} from 'rxjs';
+
+// @dynamic
+@Directive({
+    selector: '[tuiCopyProcessor]',
+})
+export class TuiCopyProcessorDirective {
+    @Input()
+    @tuiDefaultProp()
+    tuiCopyProcessor: TuiStringHandler<string> = identity;
+
+    constructor(@Inject(WINDOW) private readonly windowRef: Window) {}
+
+    @HostListener('copy.prevent', ['$event'])
+    onCopy(event: ClipboardEvent) {
+        event.clipboardData?.setData(
+            'text/plain',
+            this.tuiCopyProcessor(String(this.windowRef.getSelection())),
```

Comment:
![image](https://user-images.githubusercontent.com/12021443/145592506-615691b7-56aa-4c63-8e5c-586669465d0d.png)

what do you think?

```ts
this.tuiCopyProcessor( this.windowRef.getSelection()?.toString() ),
```

Reply:
i added check with `if` if we get null, but as for String() or toString() if there is no any difference if we have this check